modules/op-installing-tekton-hub.adoc

```diff
@@ -0,0 +1,98 @@
+
+. Clone the forked repository. 
+
+. Update the `config.yaml` file to include at least one user with the following permissions:
+* refresh a catalog
+* refresh a config file
+* create an agent token. 
++
+[source,yaml]
+----
+...
+scopes:
+- name: agent:create
+  users: <your_GitHub_username> <1>
+- name: catalog:refresh
+  users: <your_GitHub_username> <2>
+- name: config:refresh
+  users: <your_GitHub_username> <3>
+...
+----
+<1> An user with `agent:create` scope can set up a cron job that refreshes the {tekton-hub} database after an interval, if there are any changes in the catalog.
+<2> An user with `catalog:refresh` scope can refresh the catalog and all resources in the database of the {tekton-hub}.
+<3> An user with `config:refresh` scope can get additional scopes.
+
+. Commit and push the changes to your forked repository.
+
+. Create an OAuth application with your Git repository hosting provider, and note the Client ID and Client Secret. The supported service providers are GitHub, GitLab, and BitBucket.
+
+. Create the necessary formation for the {tekton-hub} API secret, which contains the following fields.
+* `GH_CLIENT_ID`: The Client ID from the GitHub OAuth application.
+* `GH_CLIENT_SECRET`: The Client Secret from the GitHub OAuth application.
+* `GHE_URL`: GitHub Enterprise URL, if you are authenticating using GitHub Enterprise. Do not provide the URL to the catalog as a value for this field.
+* `GL_CLIENT_ID`: The Client ID from the GitLab OAuth application.
+* `GL_CLIENT_SECRET`: The Client Secret from the GitLab OAuth application.
+*  `GLE_URL`: GitLab Enterprise URL, if you are authenticating using GitLab Enterprise. Do not provide the URL to the catalog as a value for this field.
+* `BB_CLIENT_ID`: The Client ID from the BitBucket OAuth application.
+* `BB_CLIENT_SECRET`: The Client Secret from the BitBucket OAuth application.
+* `JWT_SIGNING_KEY`: A long, random string used to sign the JSON Web Token (JWT) created for users.
+* `ACCESS_JWT_EXPIRES_IN`: Add the time limit after which the access token expires. For example, `1m`, where `m` denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`).
+* `REFRESH_JWT_EXPIRES_IN`: Add the time limit after which the refresh token expires. For example, `1m`, where `m` denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`). Ensure that the expiry time set for token refresh is greater than the expiry time set for token access.
+* `AUTH_BASE_URL`: Route URL for the OAuth application.
++
+[NOTE]
+====
+Use the fields related to Client ID and Client Secret for any one of the supported Git repository hosting service providers.
+==== 
+
+. Ensure that the `TektonHub` CR is similar to the following example:
++
+[source,yaml]
+----
+apiVersion: operator.tekton.dev/v1alpha1
+kind: TektonHub
+metadata:
+  name: tekton-hub-api
```

Comment:
could this be any name or must be `hub`?

Reply:
@PuneetPunamiya - Can you please comment on this?